tests/functional/aws-node-sdk/test/object/encryptionHeaders.js

```diff
@@ -167,46 +170,51 @@ describe('per object encryption headers', () => {
                     ));
 
                 testCases
-                .forEach(existing => it('should override default bucket encryption settings', done => {
-                    const _existing = Object.assign({}, existing);
-                    if (existing.masterKeyId) {
-                        _existing.masterKeyId = kmsKeyId;
-                    }
-                    const params = {
-                        Bucket: bucket,
-                        ServerSideEncryptionConfiguration: hydrateSSEConfig(_existing),
-                    };
-                    // no op putBucketNotification for the unencrypted case
-                    const s3Op = existing.algo ? (...args) => s3.putBucketEncryption(...args) : s3NoOp;
-                    s3Op(params, error => {
-                        assert.ifError(error);
-                        return putEncryptedObject(s3, bucket, object, target, kmsKeyId, error => {
+                .forEach(existing => {
+                    const hasKey = target.masterKeyId ? 'a' : 'no';
+                    const { algo } = target;
+                    it('should override bucket encryption settings with '
```

Comment:
Note to reviewer: Here I have not changed anything in the test, just improved the description a bit.